src/platform_impl/windows/event_loop.rs

```diff
@@ -1895,6 +1896,27 @@ unsafe extern "system" fn public_window_callback<T>(
             0
         }
 
+        winuser::WM_SETTINGCHANGE => {
+            use crate::event::WindowEvent::ThemeChanged;
+
+            let is_dark_mode = try_dark_mode(window);
+            let mut window_state = subclass_input.window_state.lock();
+            let changed = window_state.is_dark_mode != is_dark_mode;
+
+            if changed {
+                use crate::window::Theme::*;
+                let theme = if is_dark_mode { Dark } else { Light };
+
+                window_state.is_dark_mode = is_dark_mode;
+                subclass_input.send_event(Event::WindowEvent {
```

Comment:
```suggestion
                mem::drop(window_state);
                subclass_input.send_event(Event::WindowEvent {
```

Before sending an event we should drop `window_state`, trying to do anything on `ThemeChanged` like changing window size will otherwise crash the program.

Reply:
Good catch - committed!